### DISCUSSION TITLE
Add signal_in and signal_out to Docking Port

### DISCUSSION THREAD

**Discussion by bird-ee** [FakeFishGames/Barotrauma#8146] [Unstable feedback]
This would allow sub builders to allow the sub to distinguish between when a shuttle they have made is docked vs when a different submarine or station is docked

For instance, you could output 42 from the shuttle into the docking port, and when 42 is received by the mother sub, a different action can be taken than when no signal is received (ie a shuttle or station that the sub builder did not create)

This will make it easier to create "Twin Subs" like the Sakala community sub, where two docked halfsubs act like a singular sub when docked together, but which otherwise can dock normally when their docking target is not their sister sub. (It would be very cool to see a vanilla sub based on such a concept ^_^)

It may also simply reduce the need for wifi

**Comment by MaaiG** [comment]
Yeees!
Docking ports should be able to send and receive data when docked.
Ability to send commands to docked objects or their devices.
Submarines must be able to interact with each other and perform tasks as one.
Depending on the data type being transmitted, different logic of objects work is possible etc.

**Comment by mygamingaccount** [comment]
This would make sending signals more convenient than having a WiFi signal transmit depending on the docking port state. It would be so much more intuitive!

**Comment by mygamingaccount** [comment]
This is miscategorized as "unstable feedback" by the way

**Reply by MaaiG** [reply]
[Absorbed ))](https://github.com/Regalis11/Barotrauma/discussions/10198)
